### ISSUE TITLE
Set warnings for the ssl verify callback that is a wbemconnection parameter in python 2.7/m2crypto

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#297]
In the future pywbem  we expect to move to at least ssl and or requests package there is apparently no callback mechanism within these ssl mechanisms so that this parameter will be removed at some point in the future

**Comment by KSchopmeyer** [comment]
Added deprecated warning if this parameter is used. Also added comment about its future to the api documentation

**Comment by KSchopmeyer** [comment]
Turns out that is the ssl package used for 0.7.0 was pyopenssl, which imports as openssl.  It is specifically a wrapper layer around openssl.

Apparently Redhat moved to M2Crypto during the time they were developing for OpenLMI as part of the commits that lead to the 0.8.0 release.

Finally when we moved to release 0.8.0 and decided that we needed to support Python 3 we used the Ross Peoples code which used the SSL module (an integral module of Python releases today).

Sadly each one has some slight variations including:
1. Level of security
2. Existence of a callback function for verification of server cert.
3. exact interfaces

Note that m2Crypto and pyOpenssl are both more than just SSL connection support, they are cyptrographic packages.

However, that begs the issue of what we do now with:

    a. the fact that the verify_callback is used by M2Crypto but not used by the SSL module.  Note that the SSL module does have a callback _SSLContext.set_servername_callback()_ but is is not the same functionality as the callback in OpenSSL itself and sort of passed through to M2Crypto and PyOpenSSL python interfaces (our verify_callback parameter).

b. It appears that the level of functionality within the SSL modules varies by Python version although there appears to be some version back to Python 2.6 and Python 2.7.9 appears to have received changes to extend the functionality.  Thus, using SSL means that the actual level of security for the client will be python version dependent.

So it appears that the issue of the callback is still open, Further, depending on what we do with the next version of PyWBEM (i.e. which http and ssl package we use) we may or may not have a callback function as part of the definition.

So we can either
a. Be sure that there is a note in the documentation for 0.9.0
b. or actually use the deprecated warning.

I suggest that we back off the deprecation and just be sure we have info in the documentation that the callback is not called if Python 3 is being used.

We can clarify other issues with the next version where we should probably pick a single package for all python versions.
